### PR TITLE
kvstore.py doc strings clean up

### PR DESCRIPTION
**Pull request by madjam**

### CODE REVIEW

File: python/mxnet/kvstore.py [hunk 1]
```diff
@@ -101,16 +100,12 @@ def init(self, key, value):
...
-        1. This function returns after adding an operator to the engine.
+        """ Pushes a single or a sequence of key-value pairs into the store.
 
-        2. `push` is always called after all previous push and pull on the same
-        key are finished
-
-        3. There is no synchronization between workers. One can use ``_barrier()``
+        This function returns immeadiately after adding an operator to the engine.
+        The actual operation is executed asynchronously after all previous `push`
+        and `pull` calls for the specified input key(s) are finished.
```
Comment (piiswrong) [inline python/mxnet/kvstore.py]: the *same* input key?

**Comment by piiswrong** [comment]
@zackchase ？

File: python/mxnet/kvstore.py [hunk 2]
```diff
@@ -65,20 +65,19 @@ def __del__(self):
...
+        the value supplied by worker with rank `0` is used. This function returns
+        after data has been initialized successfully.
 
         Parameters
         ----------
         key : int or sequence of int
             The keys.
         value : NDArray or sequence of NDArray
-            The values.
+            Values corresponding to the Keys
```
Comment (zackchase) [inline python/mxnet/kvstore.py]: why is "Keys" capitalized? please lower case this

File: python/mxnet/kvstore.py [hunk 3]
```diff
@@ -119,12 +114,12 @@ def push(self, key, value, priority=0):
...
 
         value : NDArray or list of NDArray or list of list of NDArray
-            According values
+            Values corresponding to the Keys
 
         priority : int, optional
             The priority of the push operation.
-            The higher the priority, the faster this action is likely
-            to be executed before other push actions.
+            Higher priority push operations are likely to be executed before
```
Comment (zackchase) [inline python/mxnet/kvstore.py]: This is a sentence, should end in a period.

File: python/mxnet/kvstore.py [hunk 4]
```diff
@@ -167,30 +162,29 @@ def push(self, key, value, priority=0):
...
+        Subsequent attempts to read from the `out` variable will be blocked until the
+        pull operation completes.
 
-        2. `pull` is always called after all previous push and pull on the same
-        key are finished.
+        `pull` is executed asynchronously after all previous `push` and `pull` calls
+        for the same input key(s) are finished.
 
-        3. It pulls the newest value from the store.
+        The returned values are gauranteed to the latest values in the store.
```
Comment (zackchase) [inline python/mxnet/kvstore.py]: "gauranteed to the latest values in the store" what does this mean? shouldn't it be "guaranteed to be..."?

File: python/mxnet/kvstore.py [hunk 5]
```diff
@@ -167,30 +162,29 @@ def push(self, key, value, priority=0):
...
-            According values.
+            Values corresponding to the Keys.
 
         priority : int, optional
-            The priority of the push operation.
-            The higher the priority, the faster this action is likely
-            to be executed before other push actions.
+            The priority of the pull operation.
+            Higher priority pull operations are likely to be executed before
+            other pull actions
```
Comment (zackchase) [inline python/mxnet/kvstore.py]: Missing period

**Comment by madjam** [comment]
Follow up PR to address comments from @zackchase: https://github.com/dmlc/mxnet/pull/5929

**Pull request merged**